### PR TITLE
changed docker environment variables ,and subprocess settings

### PR DESCRIPTION
**Pull request by delarocamcas**
files changes:
docker-compose.template.yml = added environment variables to allow for logging printouts, and utf-8 encoded characters
dockerfile = added exiftool install on dockerfile, suppressed user prompts
requirements.txt =changed lz4 version to be compatible with setuptools
server.py = uncommented logger configuration, reformatted to use app=Bottle and app.route.
test_server.py = added in teardown method, to run after each test, cleaning up any left over test image paths.

### CODE REVIEW

File: tests/test_server.py [hunk 1]
```diff
@@ -43,6 +43,13 @@ def get_file_md5(filename):
 
 }
 
+@pytest.fixture(scope='function', autouse=True)
+def test_teardown(request):
+    try:
+        delete_attach_loc()
+    except:
+        pass
```
Comment (delarocamcas) [inline tests/test_server.py:51]: not sure if just having pass  after the except is preferable format, but not all tests post a file, so I wanted to be able to skip the delete_attach_loc without error if no file was posted.

File: server_metadata_tools.py [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
     args:
         path: path to image
         exif_dict: dictionary of exif terms using exif codes, and new values to assign"""
 
+
     exif_tag = exif_code_to_tag(exif_code)
-    command = ['exiftool', f"-{exif_tag}={exif_value}", path]
-    subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    logging.info("exif added succesfully")
+
```
Comment (delarocamcas) [inline server_metadata_tools.py:21]: added try except block around subprocess run, to print out traceback on suprocesses

File: requirements.txt
Comment (delarocamcas) [inline requirements.txt:1]: updated lz4 to be compatible with that version of setuptools

File: server.py [hunk 3]
```diff
@@ -15,26 +17,33 @@
 from datetime import datetime
 from time import sleep
 from server_metadata_tools import process_exif_ring
+from sh import convert
+from bottle import Bottle, run
+
 from image_db import ImageDb
 from image_db import TIME_FORMAT
-from sh import convert
+
```
Comment (delarocamcas) [inline server.py:25]: using app = again , to define as a Bottle app

File: server.py [hunk 4]
```diff
@@ -239,7 +247,7 @@ def resolve_file(filename, collection, type, scale):
```
Comment (delarocamcas) [inline server.py:248]: reverting to app.route

File: server.py [hunk 5]
```diff
@@ -534,21 +546,22 @@ def updatemetadata():
...
     thumb_path = path.join(thumb_root, storename)
     path_list = [orig_path, thumb_path]
-
     for rel_path in path_list:
-
         if not path.exists(rel_path):
             abort(404)
 
         if not exif_data:
             abort(400)
```
Comment (delarocamcas) [inline server.py:555]: added logger message to catch  non-dict data from being passed to the subprocesses

File: server.py [hunk 6]
```diff
@@ -15,26 +17,33 @@
...
+from bottle import Bottle, run
+
 from image_db import ImageDb
 from image_db import TIME_FORMAT
-from sh import convert
+
+app = application = Bottle()
+
+# Configure logging
+logging.basicConfig(filename='app.log', level=logging.DEBUG)
```
Comment (foozleface) [inline server.py:29]: Do you want debug level logging while in produciton?

**Review by foozleface — APPROVED** [review]
Approved, but you might not want debug level logging - your call.

**Pull request merged**